ets-demo/etstool.py

```diff
@@ -0,0 +1,361 @@
+#
+#  Copyright (c) 2017, Enthought, Inc.
+#  All rights reserved.
+#
+#  This software is provided without warranty under the terms of the BSD
+#  license included in enthought/LICENSE.txt and may be redistributed only
+#  under the conditions described in the aforementioned license.  The license
+#  is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+#  Thanks for using Enthought open source!
+#
+"""
+Tasks for Test Runs
+===================
+
+This file is intended to be used with a python environment with the
+click library to automate the process of setting up test environments
+and running the test within them.  This improves repeatability and
+reliability of tests be removing many of the variables around the
+developer's particular Python environment.  Test environment setup and
+package management is performed using `EDM
+<http://docs.enthought.com/edm/>`_
+
+To use this to run you tests, you will need to install EDM and click
+into your working environment.  You will also need to have git
+installed to access required source code from github repositories.
+You can then do::
+
+    python etstool.py install --runtime=... --toolkit=...
+
+to create a test environment from the current codebase and::
+
+    python etstool.py test --runtime=... --toolkit=...
+
+to run tests in that environment.  You can remove the environment with::
+
+    python etstool.py cleanup --runtime=... --toolkit=...
+
+If you make changes you will either need to remove and re-install the
+environment or manually update the environment using ``edm``, as
+the install performs a ``python setup.py install`` rather than a ``develop``,
+so changes in your code will not be automatically mirrored in the test
+environment.  You can update with a command like::
+
+    edm run --environment ... -- python setup.py install
```

Comment:
I think it is needed in `traits` because of the occasional need to recompile the C code, which I often just do manually.
With the `--editable` opiton in `install`, there is never a need for me to call the `update` command in any of these `etstool.py`. I often just reinstall for a clean slate.  I might be alone in that, not sure. Should I add it anyway?

Reply:
I guess I will let someone else who needs to add it.

Reply:
Personally I don't use `update` either, didn't even know it existed until I compared the docstrings. It seems that `etstool.py` in Traits is more often updated, so I thought maybe this is a nice new thing that should be brought into other `etstool.py` scripts too. But I don't have a strong opinion about this, so I'm going to leave it up to you.